crates/bevy_math/src/cubic_splines.rs

```diff
@@ -846,11 +830,287 @@ impl<P: Point> IntoIterator for CubicCurve<P> {
     }
 }
 
+impl<P: Point> RationalGenerator<P> for CubicNurbs<P> {
+    #[inline]
+    fn to_curve(&self) -> RationalCurve<P> {
+        let segments = self
+            .control_points.windows(4)
```

Comment:
In some cases, as when the knot vector has multiplicity at the ends, we may actually need to use the same control point multiple times.